substrate/utils/wasm-builder/src/wasm_project.rs

```diff
@@ -910,7 +917,10 @@ fn build_bloaty_blob(
 	if crate::build_std_required() {
 		// Unfortunately this is still a nightly-only flag, but FWIW it is pretty widely used
 		// so it's unlikely to break without a replacement.
-		build_cmd.arg("-Z").arg("build-std");
+		match target {
+			RuntimeTarget::Wasm => build_cmd.arg("-Z").arg("build-std"),
+			RuntimeTarget::Riscv => build_cmd.arg("-Z").arg("build-std=core,alloc"),
+		};
```

Comment:
Let's make `build_std_required()` return an `Option<&str>` with the required `-Zbuild-std=*` string.

Reply:
https://github.com/paritytech/polkadot-sdk/pull/6419/commits/4dd8d28e8b157f76f0cf74b0dc2b7aabfcc79c12